numpyro/infer/svi.py

```diff
@@ -93,26 +89,101 @@ class SVI(object):
+        ...     # sample latent_fairness from the distribution Beta(alpha_q, beta_q)
         ...     numpyro.sample("latent_fairness", dist.Beta(alpha_q, beta_q))
 
-        >>> data = jnp.concatenate([jnp.ones(6), jnp.zeros(4)])
+        >>> # setup the optimizer
         >>> optimizer = numpyro.optim.Adam(step_size=0.0005)
+
+        >>> # setup the inference algorithm
         >>> svi = SVI(model, guide, optimizer, loss=Trace_ELBO())
         >>> svi_result = svi.run(random.PRNGKey(0), 2000, data)
+
+        >>> # grab the learned variational parameters
         >>> params = svi_result.params
+
+        >>> # here we use some facts about the Beta distribution
+        >>> # compute the inferred mean of the coin's fairness
         >>> inferred_mean = params["alpha_q"] / (params["alpha_q"] + params["beta_q"])
+        >>> # compute inferred standard deviation
+        >>> factor = params["beta_q"] / (params["alpha_q"] * (1.0 + params["alpha_q"] + params["beta_q"]))
+        >>> inferred_std = inferred_mean * math.sqrt(factor)
+
+        >>> # sampling from guide for quick diagnostics
+        >>> samples = guide.sample_posterior(random.PRNGKey(1), params, (1000,))
+        >>> numpyro.diagnostics.print_summary(samples, prob=0.89, group_by_chain=False)
+
+        >>> # get posterior predictive samples, for arbitrary data size
+        >>> num_samples = 1000
+        >>> dist_posterior_predictive = Predictive(
+        ...     model=model,
+        ...     guide=guide,
+        ...     params=params,
+        ...     num_samples=num_samples,
+        ... )
+
+        >>> samples_posterior_predictive = dist_posterior_predictive(
+        ...     random.PRNGKey(1),
+        ...     data=data[:5],
+        ... )
+
+        >>> # get prior predictive samples, for arbitrary data size
+        >>> num_samples = 1000
+        >>> dist_prior_predictive = Predictive(
+        ...     model=model,
+        ...     params=None,
+        ...     num_samples=num_samples,
+        ... )
+
+        >>> samples_prior_predictive = dist_prior_predictive(
+        ...     random.PRNGKey(1),
+        ...     data=data[:5],
+        ... )
+
         >>> # get posterior samples
-        >>> predictive = Predictive(guide, params=params, num_samples=1000)
-        >>> samples = predictive(random.PRNGKey(1), data)
+        >>> num_samples = 1000
+        >>> dist_posterior = Predictive(
+        ...     model=guide,
+        ...     params=params,
+        ...     num_samples=num_samples,
+        ... )
+        >>> samples_posterior = dist_posterior(random.PRNGKey(1))
+
+        >>> # get prior samples, for full data
+        >>> num_samples = 1000
+        >>> dist_prior = Predictive(
+        ...     model=guide,
+        ...     params=None,
+        ...     num_samples=num_samples,
+        ... )
+
+        >>> samples_prior = dist_prior(
+        ...     random.PRNGKey(1),
+        ...     data=data,
+        ... )
```

Comment:
I think it is better to remove either `full data` part or partial data (should we use new data rather than partial data) part. Actually, I'm a bit confused of several notions here. What posterior predictive with arbitrary data size does? I think we need to set `data=None` for posterior predictive. Could we update the model for that pattern and only include code to show:
+ how to get posterior samples from the guide
+ (optional because it is not relevant to SVI and we already explain it in [bayesian regression](https://num.pyro.ai/en/stable/tutorials/bayesian_regression.html#Prior-Predictive-Distribution) tutorial) how to get prior samples (having data or not is not important to get prior samples - if we want to get prior predictive, it is necessary to set data=None)
+ how to use the learned parameters to draw predictive samples (with data=None): there are two ways to do this:
  + using posterior_samples above
  + using guide directly